vp/workflow/views.py

```diff
@@ -103,27 +105,72 @@ def save_workflow(request):
         'message': 'Saved the graph to ' + workflow.file_path + '!'
     })
 
-def retrieve_nodes_for_user(request):
-    if request.method =='GET':
+def retrieve_nodes_for_ide(request):
+    if request.method == 'GET':
+        """Retrieve all nodes that a user can have access to in the IDE from a specified file in request body.
+        
+        Args:
+            request: Django request Object
+       
+         Returns:
+            200 - JSON response with data.
+            400 - No file specified/issues reading json file.
+            404 - File specified not found, or not JSON graph.
         """
-        Retrieve all nodes that a user can have access to in the IDE.
-        Currently returning default set of nodes. 
-        //TODO pick these node files from a file in the system.
-        """
-        data = {
-            "I/O": [
-                {"key": "read-csv", "name": "Read CSV", "numPortsIn": 0, "color": "black"}
-            ],
-            "Manipulation": [
-                {"key": "filter", "name": "Filter Rows", "color": "red"},
-                {"key": "pivot", "name": "Pivot Table", "color": "blue"},
-                {"key": "multi-in", "name": "Multi-Input Example", "numPortsIn": 3, "color": "green"}
-            ]
-        }
+        try:
+            request_body = json.loads(request.body)
+            file_to_load = request_body['file']
+            if file_to_load is None:
+                return JsonResponse({'message': 'Please specify nodes file.'}, status=400)
+            elif not os.path.exists(file_to_load):
+                return JsonResponse({'message': 'Unable to locate nodes file.'}, status=404)
+            f = open(file_to_load)
+            data = json.load(f)
```

Comment:
Should change to `with open(file_to_save) as f`. Removes the need for the `else: f.close()` because [Python provides automatic file handling when using `with`](https://docs.python.org/3/tutorial/inputoutput.html#reading-and-writing-files). Also, I believe it should catch [`OSError`](https://docs.python.org/3/library/functions.html#open) instead of `IOError`.

Reply:
That's right, added this fix. Thanks!